giraph-core/src/main/java/org/apache/giraph/worker/BspServiceWorker.java

```diff
@@ -1236,6 +1233,57 @@ public void cleanup(FinishedSuperstepStats finishedSuperstepStats)
       LOG.error("cleanup: Got InterruptedException on notification " +
           "to master about cleanup", e);
     }
+  }
+
+  /**
+   * Method to send the counter values to the master using the zookeeper
+   * This is called after finishing all the supersteps
+   *
+   */
+  public void sendCountersToMaster() {
```

Comment:
nit: since this is really storing counters in ZK, and not sending to master directly, let's change the name to reflect this.